src/Routing/Dispatcher.php

```diff
@@ -50,19 +50,19 @@ class Dispatcher
      * If the controller is found, and the action is not found an exception will be thrown.
      *
      * @param \Cake\Http\ServerRequest $request Request object to dispatch.
-     * @param \Cake\Network\Response $response Response object to put the results of the dispatch into.
+     * @param \Cake\Http\Response $response Response object to put the results of the dispatch into.
      * @return string|null if `$request['return']` is set then it returns response body, null otherwise
      * @throws \LogicException When the controller did not get created in the Dispatcher.beforeDispatch event.
      */
     public function dispatch(ServerRequest $request, Response $response)
     {
         $actionDispatcher = new ActionDispatcher(null, $this->eventManager(), $this->_filters);
         $response = $actionDispatcher->dispatch($request, $response);
-        if (isset($request->params['return'])) {
-            return $response->body();
+        if (!isset($request->params['return'])) {
```

Comment:
See the refactoring I tried. But since the void is a lie I had to Abort. I can revert then now.